DOCS-3867: TrueCommand 2.2 & SCALE Clustering Documentation (#1424)
truenas/documentation: content/TrueCommand/Clusters.md

---
title: "Clusters Screen"
description: "Descriptions of the options contained inside the TrueCommand Clusters menu."
weight: 60
geekdocCollapseSection: true
tags:
 - scaleclustering
aliases:
 - /truecommand/clustering/
---

{{< toc >}}

TrueCommand 2.2, in conjunction with TrueNAS SCALE, can create and manage clusters, cluster volumes, and cluster volume shares.

{{< hint danger >}}
Clusters are an experimental feature in TrueCommand 2.2.
Before using such features, please back up all your data.
Do not rely on this for critical data.
{{< /hint >}}

## Clusters

The **Clusters** screen contains all options related to the cluster feature.
To see this screen, log in to TrueCommand and click the <span class="iconify" data-icon="mdi:server-network"></span> **Clusters** icon in the upper left.

![ClustersScreen](/images/TrueCommand/2.2/ClustersScreen.png "TrueCommand Clusters Screen")

{{< expand "Option descriptions (click to expand)" "v" >}}
When no clusters are created, there is a single option on this screen:

> | Setting | Value | Description |

Reviewer: remove the Value column to be consistent with the CORE and SCALE documentation. Or we can do that later when we reorg TC content.

|---------|-------|-------------|

> | CREATE CLUSTER | button | Opens the form to create a new cluster. |

Reviewer: suggest removing the third table column.
```suggestion
| **CREATE CLUSTER** | button | Opens the form to create a new cluster. |
```

{{< /expand >}}

### Create Cluster

The cluster creation options are split into two pages: system selection and confirmation.

#### Systems

The initial form has two fields:

![CreateClusterSystems](/images/TrueCommand/2.2/CreateClusterSystems.png "Configuring Systems in the Cluster")

{{< expand "Option descriptions (click to expand)" "v" >}}

> | Setting | Value | Description |

Reviewer: suggest removing the third table column for consistency with CORE and SCALE docs
```suggestion
| Setting | Description |
```

|---------|-------|-------------|

> | Name | string | Identifies the cluster. |

Reviewer: ```suggestion
| **Name** |  Enter a string to identify the cluster. |
```

> | Systems | dropdown | Shows all connected TrueNAS SCALE systems available for this cluster. Clusters can consist of between 3 and 20 individual SCALE systems. |

Reviewer: ```suggestion
| **Systems** | Dropdown shows all connected TrueNAS SCALE systems available for this cluster. Clusters can consist of between 3 and 20 individual SCALE systems. |
```

| Next | button | Advances the form to the next screen. |

{{< /expand >}}

Making selections in the **Systems** field adds more options.

![CreateClusterSystemsNetwork](/images/TrueCommand/2.2/CreateClusterSystemsNetwork.png "Network Options for Clustered Systems")

{{< expand "Option descriptions (click to expand)" "v" >}}

| Setting | Value | Description |

> |---------|-------|-------------|

Reviewer: ```suggestion
|---------|-------------|
```

> | Delete | button | Clicking the  <i class="material-icons" aria-hidden="true" title="Delete">delete</i> **Delete** icon removes the system from the cluster. |

Reviewer: ```suggestion
| **Delete** | Clicking the  <i class="material-icons" aria-hidden="true" title="Delete">delete</i> **Delete** icon removes the system from the cluster. |
```

> | Network Address | dropdown | Shows available IP addresses to use for cluster traffic. Using private dedicated network addresses is recommended. |

Reviewer: ```suggestion
| **Network Address** | Dropdown list shows available IP addresses to use for cluster traffic. Using private dedicated network addresses is recommended. |
```

{{< /expand >}}

#### Confirmation

There is a single option on this page.

![CreateClusterReview](/images/TrueCommand/2.2/CreateClusterReview.png "Review and create page for Clusters")

{{< expand "Option descriptions (click to expand)" "v" >}}

> | Setting | Value | Description |

Reviewer: ```suggestion
| Setting | Description |
```

> |---------|-------|-------------|

Reviewer: ```suggestion
|---------|-------------|
```

> | CREATE | button | Begins the creation process for a cluster. This restricts the SMB functionality on the SCALE systems. |

Reviewer: ```suggestion
| **CREATE** | Begins the creation process for a cluster. This restricts the SMB functionality on the SCALE systems. |
```

{{< /expand >}}

### Configure Cluster

Successfully creating a cluster immediately adds a cluster widget to the **Clusters** screen and opens options to configure the new cluster.
These options are split into three screens: **SMB Network**, **Active Directory**, and **Confirmation**.

#### SMB Network

The SMB Network page has general options and fields added for each SCALE system in the cluster.

![ConfigureClusterSMBNetwork](/images/TrueCommand/2.2/ConfigureClusterSMBNetwork.png "Configure Cluster SMB Network")

{{< expand "Option descriptions (click to expand)" "v" >}}

> | Setting | Value | Description |

Reviewer: ```suggestion
| Setting | Description |
```

> |---------|-------|-------------|

Reviewer: ```suggestion
|---------|-------------|
```

> | NEXT | button | Continues to the next page. |

Reviewer: ```suggestion
| **NEXT** | Continues to the next page. |
```

> | Skip this step | checkbox | Finalize the cluster configuration without registering a network to use for SMB sharing. Setting this permanently removes the ability to share data stored in this cluster. |

Reviewer: ```suggestion
| **Skip this step** | Select to finalize the cluster configuration without registering a network to use for SMB sharing. This permanently removes the ability to share data stored in this cluster. |
```

> | Interface | dropdown | SCALE system network interface to use for SMB traffic. For best functionality, choose an interface that can communicate between the individual SCALE systems, TrueCommand environment, Active Directory, and and client systems that need to access data shared from this cluster. |

Reviewer: ```suggestion
| **Interface** | SCALE system network interface to use for SMB traffic. For best functionality, select an interface from the dropdown list that can communicate between the individual SCALE systems, TrueCommand environment, Active Directory, and and client systems that need to access data shared from this cluster. |
```

> | Address | dropdown | IP address used for clients to access the SCALE system. |

Reviewer: ```suggestion
| **Address** | Select the IP address from the dropdown list used for clients to access the SCALE system. |
```
